src/Components/Components/src/RenderTree/EventHandlerDelegateWithClosuresComparer.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Linq.Expressions;
+
+namespace Microsoft.AspNetCore.Components.RenderTree
+{
+    internal static class EventHandlerDelegateWithClosuresComparer
+    {
+        private static readonly ConcurrentDictionary<Type, Func<object, object, bool>> ClosureComparerCache = new();
+
+        internal static bool EventCallBackEquals(ref EventCallback left, ref EventCallback right)
+        {
+            if (Equals(left, right))
+            {
+                return true;
+            }
+
+            // Whenever the receivers are different and they are explicit we assume that
+            // the EventCallbacks are not equal.
```

Comment:
Would it be possible to replace this comment with one that says *why* we're doing this (and why we believe it to be valid) rather than saying *what* the code does?

I'm sure you have a good reason for this condition based on `RequiresExplicitReceiver`, but it's not clear to me at first glance, and it would be even less clear to someone who had never heard about `RequiresExplicitReceiver` before :)

Reply:
In particular it's unclear to me why we test for `left.Receiver != right.Receiver` before testing `left.RequiresExplicitReceiver || right.RequiresExplicitReceiver`, based on how `RequiresExplicitReceiver` is defined.

Reply:
Normally an EventCallback is equal when its members are equal, i.e. Receiver and Delegate are equal. Now there is a special case, where the delegates can point to different closure instances but are equal anyway. We have to test for that. However, when the receivers differ (i.e. they point to a different component) and they are explicit (i.e. they are not the same as the delegates targets, because then they could be closures again) we know that they are different. O